model/dataProvider/providers/ResultsDataProviderService.php

```diff
@@ -0,0 +1,138 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoSyncClient\model\dataProvider\providers;
+
+use common_exception_NotFound;
+use core_kernel_persistence_Exception;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoDeliveryRdf\helper\DetectTestAndItemIdentifiersHelper;
+use oat\taoResultServer\models\classes\ResultManagement;
+use oat\taoResultServer\models\classes\ResultServerService;
+use oat\taoSyncClient\model\dataProvider\SyncPackageDataProviderInterface;
+use taoResultServer_models_classes_ReadableResultStorage;
+use taoResultServer_models_classes_WritableResultStorage;
+
+/**
+ * Class ResultDataProviderService
+ * @package oat\taoSyncClient\model\dataProvider\providers
+ */
+class ResultsDataProviderService extends ConfigurableService implements SyncPackageDataProviderInterface
+{
+    /**
+     * @param array $deliveryExecutionIds
+     * @return array
+     * @throws common_exception_NotFound
+     * @throws core_kernel_persistence_Exception
+     */
+    public function getData($deliveryExecutionIds = [])
+    {
+        $results = [];
+        foreach ($deliveryExecutionIds as $deliveryExecutionId) {
+            /** @var DeliveryExecution $deliveryExecution */
+            $deliveryExecution = $this->getServiceProxy()->getDeliveryExecution($deliveryExecutionId);
+            $variables = $this->getDeliveryExecutionVariables($deliveryExecution->getDelivery()->getUri(), $deliveryExecutionId);
+            $results[] = [
+                'deliveryId'          => $deliveryExecution->getDelivery()->getUri(),
+                'deliveryExecutionId' => $deliveryExecutionId,
+                'details'             => $this->getDeliveryExecutionDetails($deliveryExecutionId),
+                'variables'           => $variables,
+            ];
+        }
+        return $results;
+    }
+
+    /**
+     * Get details of a delivery execution
+     *
+     * @param $deliveryExecutionId
+     * @return array
+     */
+    private function getDeliveryExecutionDetails($deliveryExecutionId)
+    {
+        /** @var DeliveryExecution $deliveryExecution */
+        $deliveryExecution = $this->getServiceProxy()->getDeliveryExecution($deliveryExecutionId);
+        try {
+            return [
+                'identifier' => $deliveryExecution->getIdentifier(),
+                'label'      => $deliveryExecution->getLabel(),
+                'test-taker' => $deliveryExecution->getUserIdentifier(),
+                'starttime'  => $deliveryExecution->getStartTime(),
+                'finishtime' => $deliveryExecution->getFinishTime(),
+                'state'      => $deliveryExecution->getState()->getUri(),
+            ];
+        } catch (common_exception_NotFound $e) {
+            return [];
```

Comment:
Do you think it's ok to return empty array if we cannot find delivery execution details or it would be better to do not add this result into a synchronization package? Will we be able to import result without delivery execution details?

Reply:
now it is exception and synchronization will be failed.